miditok/tokenizations/remi_plus.py

```diff
@@ -0,0 +1,814 @@
+        else:
+            candidates = cls.__get_candidates(chroma=chroma)
+            scores, qualities = cls.__get_score(candidates=candidates)
+            # bass note
+            sorted_notes = []
+            for i, v in enumerate(np.sum(pianoroll, axis=0)):
+                if v > 0:
+                    sorted_notes.append(int(i % 12))
+            bass_note = sorted_notes[0]
+            # root note
+            __root_note = []
+            _max = max(scores.values())
+            for _root_note, score in scores.items():
+                if score == _max:
+                    __root_note.append(_root_note)
+            if len(__root_note) == 1:
+                root_note = __root_note[0]
+            else:
+                for n in sorted_notes:
+                    if n in __root_note:
+                        root_note = n
+                        break
+                return "None", "None", "None", 0  # no root found
+            # quality
+            quality = qualities.get(root_note, "None")
+            sequence = candidates.get(root_note, [])
+            # score
+            score = scores.get(root_note, 0)
+            return (
+                _PITCH_CLASSES[root_note],
+                quality,
+                _PITCH_CLASSES[bass_note],
+                score,
+            )
+
+    @classmethod
+    def __solve(
+        cls,
+        candidates: Dict[int, Dict[int, Tuple[int, float, int, float]]],
+        max_tick: int,
+    ) -> List[Tuple[int, int, str]]:
+        chords: List[Tuple[int, int, str]] = []
+        start_tick = 0
+        while start_tick < max_tick:
+            _candidates = candidates.get(start_tick, {})
+            _candidates = sorted(_candidates.items(), key=lambda x: (x[1][-1], x[0]))
+            # choose
+            end_tick, (root_note, quality, bass_note, _) = _candidates[-1]
+            if root_note == bass_note:
+                chord = "{}:{}".format(root_note, quality)
+            else:
+                chord = "{}:{}/{}".format(root_note, quality, bass_note)
+            chords.append((start_tick, end_tick, chord))
+            start_tick = end_tick
+        # remove :None
+        __temp = copy(chords)
+        while ":None" in str(__temp[0][-1]):
+            try:
+                _new_head = (__temp[0][0], __temp[1][1], __temp[1][2])
+                del __temp[0]  # delete None
+                __temp = [_new_head] + __temp[1:]
+            except:
+                return []
+        __temp2 = []
+        for chord in __temp:
+            if ":None" not in str(chord[-1]):
+                __temp2.append(chord)
+            else:
+                # __temp2[-1][1] = chord[1]
+                __temp2 = __temp2[:-1] + [(__temp2[-1][0], chord[1], __temp2[-1][2])]
+        return __temp2
+
+    @classmethod
+    def extract(cls, notes: List[Note]) -> List[Tuple[int, int, str]]:
+        # read
+        max_tick = max([n.end for n in notes])
```

Comment:
Note start and end attributes are float type, even if in practice they are always int values.
Should maybe add `int()` just in case, which would remove the warning in `range` lines 230 and 256.